pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -622,30 +749,29 @@ func (s *DiscoveryServer) pushEds(push *model.PushContext, con *XdsConnection,
 	emptyClusters := 0
 	endpoints := 0
 	empty := []string{}
+	sidecarScope := con.modelNode.SidecarScope
 
+	// All clusters that this endpoint is watching. For 1.0 - it's typically all clusters in the mesh.
+	// For 1.1+Sidecar - it's the small set of explicitly imported clusters, using the isolated DestinationRules
 	for _, clusterName := range con.Clusters {
 
 		_, _, hostname, _ := model.ParseSubsetKey(clusterName)
 		if edsUpdatedServices != nil && edsUpdatedServices[string(hostname)] == nil {
-			// Cluster was not updated, skip recomputing.
+			// Cluster was not updated, skip recomputing. This happens when we get an incremental update for a
+			// specific Hostname. On connect or for full push edsUpdatedServices will be empty.
 			continue
 		}
 
-		c := s.getEdsCluster(clusterName)
-		if c == nil {
-			totalXDSInternalErrors.Add(1)
-			adsLog.Errorf("cluster %s was nil skipping it.", clusterName)
-			continue
+		var l *xdsapi.ClusterLoadAssignment
+		var ok bool
+		// decide which to use based on presence of Sidecar.
+		if sidecarScope == nil || sidecarScope.Config == nil {
```

Comment:
per earlier comment, you dont need this. Also, the destination rule bug scenario (local namespace first, then namespace of the service used in the dest rule) applies to 1.0. So the selection logic actually applies to both new and old, since we treated the destination rule thing as a "bug" fix.